homeassistant/components/fan/__init__.py

```diff
@@ -241,13 +248,35 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
         """Set new preset mode."""
         await self.hass.async_add_executor_job(self.set_preset_mode, preset_mode)
 
+    @callback
     def _valid_preset_mode_or_raise(self, preset_mode: str) -> None:
-        """Raise NotValidPresetModeError on invalid preset_mode."""
+        """Raise ServiceValidationError on invalid preset_mode."""
+        _LOGGER.warning(
```

Comment:
Besides the logging should we not create an issue?

Reply:
maybe